llvm-hs/src/LLVM/Internal/FFI/MetadataC.cpp

```diff
@@ -146,5 +147,27 @@ void LLVM_Hs_MetadataReplaceAllUsesWith(LLVMMetadataRef md, LLVMMetadataRef repl
     MDNode::deleteTemporary(Node);
 }
 
+LLVMMetadataRef LLVM_Hs_IsADILocation(LLVMMetadataRef md) {
+    if (isa<DILocation>(unwrap(md))) {
+        return md;
+    }
+    return nullptr;
+}
+
+unsigned LLVM_Hs_GetMetadataClassId(LLVMMetadataRef md) {
```

Comment:
If I can find a way to extract the mapping of `ID -> Class` from the C++ this could be used during decoding.